docs: Migrate thermocycler instruction manual to MkDocs

jwwojak opened the pull request:
# Overview

This migrates content from the Thermocycler GEN2 Instruction Manual to markdown and mkdocs.

Most of the content is the same as the manual. Some new changes particularly around the installation procedures and different lid types. Remainder is largely the same.

RTC-780, RTC-710

## Test Plan and Hands on Testing

Please check for the usual writing and code stuff.

## Changelog

- Flex and OT-2 installation instructions now separate .md files.
- Updated lid info, including warnings about using rubber seals and PCR lids at the same time.
- Adds information about the new shipping insert that replaces the rubber seal.

## Review requests

I no longer have many of the original images. These may be in .indd files. Any problems with images, pull originals from the InDesign files. I do not have this software.

## Risk assessment

Docs changes, low risk.

jwwojak committed:
Create the new project

jwwojak committed:
Update mkdocs.yml

jwwojak committed:
quick save

jwwojak committed:
Mid-day commit

jwwojak committed:
Rebase and updated

Got latest code for docs

Updated to add a module card for thermocycler

Updated to add TOC entry for thermocycler on modules page and link to manual.

jwwojak committed:
Compliance section save

Text and warning images.

jwwojak committed:
Safety and compliance section commit

jwwojak committed:
Start product specs section

jwwojak committed:
Specifications section

Clipped images from manual.

Do not have individual images. Don't look too bad, but we should try to get originals from .indd files. I don't have InDesign and can't do this.

Using placeholder images to complete doc.

jwwojak committed:
End 'o day commit

Getting .yml file key-values sorted to use the page title feature.

jwwojak committed:
More changes

jwwojak committed:
Midday save

jwwojak committed:
Migration and conversion commit

Finished the converting and getting it all together phase, from PDF to markdown.

jwwojak committed:
Finished migration and changes

Opening PR

jwwojak committed:
Image name changes and missing image files

Fixed file names and added a missing image.

ecormany commented on docs/thermocycler-manual/docs/compliance.md:
```diff
@@ -0,0 +1,187 @@
+
+| Environmental Conditions | Recommended | Acceptable | Storage and Transportation |
+|----|----|----|----|
+| Ambient Temperatures | +20 to +25 °C | +2 to +40 °C | -10 to +60 °C |
```
Punctuation nitpick (minus sign).
```suggestion
| Ambient Temperatures | +20 to +25 °C | +2 to +40 °C | –10 to +60 °C |
```

ecormany commented on docs/thermocycler-manual/docs/compliance.md:
```diff
@@ -0,0 +1,187 @@
+---
+title: "Thermocycler GEN2 Instruction Manual: Safety and Compliance"
```
I think using a short product name in these headers, rather than the full publication name, looks nicer and is easier to read.
```suggestion
title: "Thermocycler Module: Safety and Compliance"
```

ecormany commented on docs/thermocycler-manual/docs/compliance.md:
```diff
@@ -0,0 +1,187 @@
+|----|----|
+| Recommended | Opentrons has validated the Thermocycler’s performance in the conditions recommended for system operation. Operating the Thermocycler in these conditions helps provide optimal results. |
+| Acceptable | The Thermocycler is safe to use in conditions acceptable for system operation,
+but results may vary. |
```
Extra newline here breaks table.
```suggestion
| Acceptable | The Thermocycler is safe to use in conditions acceptable for system operation, but results may vary. |
```

ecormany commented on docs/thermocycler-manual/docs/compliance.md:
```diff
@@ -0,0 +1,187 @@
+
+!!!warning
+    Do not set the plate temperature below
+the ambient temperature for over two hours, as the resulting condensation could affect performance or damage the module.
```
Same here, the newline causes this to spill out of the warning.
```suggestion
    Do not set the plate temperature below the ambient temperature for over two hours, as the resulting condensation could affect performance or damage the module.
```

ecormany commented on docs/thermocycler-manual/docs/compliance.md:
```diff
@@ -0,0 +1,187 @@
+### Safety
+
+- IEC/UL/CSA 61010-1 Safety­ Requirements­ for­ Electrical Equipment­ for Measurement­ Control­ and­ Laboratory Use–­Part­ 1:­ General Requirements
+- IEC/UL/CSA ­ 61010-2-010­ Particular­ Requirements­ for Laboratory­ Equipment for­ the Heating­ of­ Materials
```
Extra space (which is, surprisingly, preserved in the HTML output!)
```suggestion
- IEC/UL/CSA­ 61010-2-010­ Particular­ Requirements­ for Laboratory­ Equipment for­ the Heating­ of­ Materials
```

ecormany commented on docs/thermocycler-manual/docs/compliance.md:
```diff
@@ -0,0 +1,187 @@
+- EN/BSI 61326-1­ Electrical ­ Equipment­ for Measurement Control­ and­ Laboratory­ Use­–EMC Requirements–Part­ 1: General­ Requirements
+- EN 55011 Industrial, Scientific and Medical Equipment–Radio Frequency­ Disturbance Characteristics–Limits­ and­ Methods of­ Measurement
+- FCC 47CFR Part 15 Subpart B Class A: Unintentional Radiators
+- IC ICES–003 Spectrum Management and Telecommunications Interference Causing Equipment Standard–Information Technology Equipment (Including Digital Apparatus)
```
The rare en dash that should be a hyphen!
```suggestion
- IC ICES-003 Spectrum Management and Telecommunications Interference Causing Equipment Standard–Information Technology Equipment (Including Digital Apparatus)
```

jwwojak replied:
At least it isn't a semi-colon :wink:

ecormany commented on docs/thermocycler-manual/docs/compliance.md:
```diff
@@ -0,0 +1,187 @@
+
+### Canada ISED
+
+Canada ICES–003(A)/NMB–003(A)
```
Hyphens here too.
```suggestion
Canada ICES-003(A)/NMB-003(A)
```

ecormany commented on docs/thermocycler-manual/docs/specifications.md:
```diff
@@ -0,0 +1,110 @@
+      <td>
+        <ul>
+          <li>Range: 4–99 °C</li>
+          <li>Maximum heating ramp rate: 4.25 °C/s from ambient to 95 °C to ambient</li>
```
Lost some text here.
```suggestion
          <li>Maximum heating ramp rate: 4.25 °C/s from ambient to 95 °C</li>
          <li>Maximum cooling ramp rate: 2.0 °C/s from 95 °C to ambient</li>
```

ecormany commented on docs/thermocycler-manual/docs/specifications.md:
```diff
@@ -0,0 +1,110 @@
+  </thead>
+  <tbody>
+    <tr>
+      <td>Solid white</td>
```
We could improve these with our colored dots in the future, but this accurately reflects the current PDF manual. Make a ticket if you want to come back to this after this PR.

jwwojak replied:
I am trying this now. It's a great suggestion.

We will have to add some new code to the `instruction-manual.css` file too. I need to make a flex box to get the dot and corresponding text to line up properly. This will be `.status-dot-container` at the end of all the status dot code.

➕  moving the important warning at the _end_ of the table into the first row, as it belongs there (a warning that in a white idle state the module can still be hot). The admonition won't work b/c this is HTML, but I have a unicode warning label ⚠️ instead.

ecormany commented on docs/thermocycler-manual/mkdocs.yml:
```diff
@@ -0,0 +1,59 @@
+  - Product Description: index.md
+  - Safety and Compliance: compliance.md
+  - Product Specifications: specifications.md
+  - Before You Begin: preinstall.md
```
I think it makes more sense to put this as a child of the "Installing the Thermocycler" section. That bit of new organization is really helpful.

jwwojak replied:
Moved into the installation section.

ecormany commented on docs/thermocycler-manual/docs/preinstall.md:
```diff
@@ -0,0 +1,53 @@
+
+## Flex Caddies
+
+When used with a Flex robot, the Thermocycler fits into a caddy that occupies space below the deck. The caddy places your labware closer to the deck surface and allows for below-deck cable routing. See the Modules chapter in the Flex Instruction Manual for more information.
```
We can link this now! 🎉 
```suggestion
When used with a Flex robot, the Thermocycler fits into a caddy that occupies space below the deck. The caddy places your labware closer to the deck surface and allows for below-deck cable routing. See the [Modules chapter](../flex-manual/modules.md) in the Flex Instruction Manual for more information.
```

ecormany commented on docs/thermocycler-manual/docs/flex-installation.md:
```diff
@@ -0,0 +1,66 @@
+The tools you need are included in the User Kit that came with
+your Flex.
+
+## Attaching the Thermocycler
```
Let's wrap these steps in a `<div class="instruction-list">` to get the fancy bullet formatting.

jwwojak replied:
Added the `instruction-list` class to the procedure steps.

ecormany commented on docs/thermocycler-manual/docs/flex-installation.md:
```diff
@@ -0,0 +1,66 @@
+!!!warning
+    The gantry and pipette move during calibration. Keep your hands clear of the working area before tapping an action button on the touchscreen.
+
+1. Tap **Start setup** on the touchscreen. The robot checks the
```
We have fancy bullets on these steps in the PDF, but I kind of like leaving them plain.

jwwojak replied:
Oh, I applied the style to this. The fancy bullets might make things look crowded. Take a look. Can revert back.

ecormany replied:
Looks fine, no need to revert.

ecormany commented on docs/thermocycler-manual/docs/ot2-installation.md:
```diff
@@ -0,0 +1,27 @@
+title: "Thermocycler GEN2 Instruction Manual: OT-2 Attachment Steps"
+---
+
+# OT-2 Attachment Steps
```
Add instruction list formatting in this section too.

jwwojak replied:
✔️ Will do.

ecormany commented on docs/thermocycler-manual/docs/ot2-installation.md:
```diff
@@ -0,0 +1,27 @@
+
+1. Use a 2.5 mm hex screwdriver or L-key to attach the exhaust duct to the back of the module. Use the two included screws, one on either side of the duct.
+
+![Attaching the exhaust duct](../images/ot2-exhaust-duct.png)
```
Probably best to indent this four spaces so the image is rendered as part of step 1.
```suggestion
    ![Attaching the exhaust duct](../images/ot2-exhaust-duct.png)
```

ecormany commented on docs/thermocycler-manual/docs/flex-installation.md:
```diff
@@ -0,0 +1,66 @@
+
+10. Press the illuminated button on the Thermocycler to open the lid.
+
+11. Remove and discard the shipping insert after opening the module's lid.
```
The OT-2 equivalent of this step has an additional sentence: "The insert consists of a pad attached to a PCR plate." Add here? Or delete there. I don't have a strong opinion as long as it's consistent.

jwwojak replied:
Yes, will include and make sure it is in the Flex and OT-2 instructions.

ecormany commented on docs/thermocycler-manual/docs/index.md:
```diff
@@ -0,0 +1,27 @@
+
+**Opentrons Labworks Inc.**
+
+December 2024
```
The lid and seal stuff is new. Should we go ahead and update to the current date?

jwwojak replied:
Updating that and all copyright stuff to "August 2025"

ecormany reviewed (commented):
Nice, looks really good! Only a few small copy and formatting fixes to make before this is ready to publish.

jwwojak committed:
Apply suggestions from code review

Making batch suggestions. Will pull changes and make additional changes locally and then push.

Co-authored-by: Ed Cormany <edward.cormany@opentrons.com>

jwwojak committed:
Revisions from reviewer comments

- change frontmatter titles
- add the instruction-list class to apply styling to bullets
- fixing LED table. See comments in the PR for details.
- update copyright to August 2025
- save now just in case, more TK today

jwwojak reviewed (commented):
Making changes from reviewer input.

jwwojak committed:
Final changes

Last changes, checking for other revisions throughout. Lot's of small things that aren't really needed, but could help.

ecormany commented on docs/thermocycler-manual/docs/compliance.md:
```diff
@@ -24,12 +24,12 @@ The Thermocycler has the following power input requirements, which are met by th
 ### Environmental Conditions { #thermocycler-environmental-conditions }
 
-The Thermocycler should only be used indoors on a sturdy, dry, flat horizontal surface. It must be installed in a low-vibration environment with stable ambient conditions. Keep the Thermocycler away from direct sunlight or HVAC systems that may cause significant temperature or humidity changes.
+The Thermocycler should only be used indoors on a sturdy, dry, flat horizontal surface. Install this module in a low-vibration environment with stable ambient conditions. Keep it away from direct sunlight or HVAC systems that may cause significant temperature or humidity changes."
```
Delete stray quotation mark

jwwojak replied:
Yikes. Removing " mark.

ecormany commented on docs/thermocycler-manual/docs/compliance.md:
```diff
@@ -47,22 +47,22 @@ The following table lists and defines standards for recommended use, acceptable
 ### Low Temperature Condensation
 
-Setting the Thermocycler to hold a temperature below the ambient temperature for a long time can cause condensation to develop in and on the device.
+Holding the Thermocycler below ambient temperature for extended periods can cause condensation to develop in and on the device
```
Missing period at end of sentence.

jwwojak replied:
📃 fixing that.

ecormany commented on docs/thermocycler-manual/docs/compliance.md:
```diff
@@ -47,22 +47,22 @@ The following table lists and defines standards for recommended use, acceptable
 ### Software Requirements
 
-The Thermocycler requires version 6.2.0 (or higher) of the Opentrons App and robot software. You can download the Opentrons App for Mac, Windows, or Linux at <https://opentrons.com/ot-app>.
+The Thermocycler requires Opentrons App and robot software version 6.2.0 or higher, available for Mac, Windows, or Linux at <https://opentrons.com/ot-app>.
```
The Flex robot software is not available directly at that link. But this is probably fine. You get robot software updates via the app.

jwwojak replied:
🤔 good point. 

will go back to the previous version.

ecormany commented on docs/thermocycler-manual/docs/compliance.md:
```diff
@@ -73,75 +73,31 @@ Warning symbols posted on the Opentrons Thermocycler and in this manual warn you
-      <td><strong>CAUTION: Warning about risk of burning.</strong><br>
-          The Opentrons Thermocycler generates enough heat to cause serious burns. Wear safety goggles or other eye protection at all times during operation. Always ensure the sample block returns to idle temperature before opening the lid and removing samples. Always allow maximum clearance to avoid accidental burns. The seal on the lid can also be hot when opened. Unplug the unit after use, if  ...
+      <td><strong>CAUTION: Hot surface.</strong><br>
+          The Opentrons Thermocycler generates enough heat to cause serious burns. Wear safety goggles or other eye protection at all times during operation. Ensure the sample block returns to idle temperature before opening the lid and removing samples. Allow maximum clearance to avoid accidental burns. The lid seal can also be hot when opened. Unplug the unit after use, if possible.
```
"Always" is strengthening in warnings like this. I would at least keep it on the first one: "Always ensure the sample block…"

jwwojak replied:
Fixing that.

ecormany commented on docs/thermocycler-manual/docs/lid-seals.md:
```diff
@@ -30,24 +30,24 @@ The Opentrons auto-sealing lids work with the following Flex hardware and softwa
 ## Reusable Automation Seals (GEN2)
 
-While disposable PCR lids offer sterile, single-use convenience, the Thermocycler GEN2 also ships with a package of reusable automation seals. Unlike the disposable lids, these seals need to be sanitized before use and can be reused up to 20 times. Note, too, that only one type of seal (disposable or reusable) should be used at a time; you must not use both on the same thermocycler simultaneously ...
+While disposable PCR lids offer sterile, single-use convenience, the Thermocycler GEN2 also ships with a package of reusable automation seals. Unlike the disposable lids, these seals require sanitization before use and can be reused up to 20 times. Note that only one type of seal (disposable or reusable) should be used at a time; _do not_ use both simultaneously on the same thermocycler.
```
Capitalize "Thermocycler" (final word of this line).

jwwojak replied:
Fixing, adding capitalization.

ecormany commented on docs/thermocycler-manual/docs/warranty.md:
```diff
@@ -10,7 +10,7 @@ All hardware purchased from Opentrons is covered under a 1-year standard warrant
 Opentrons Support can help you with questions about our products and services. If you discover a defect, or believe your product is not functioning to published specifications, contact us at <support@opentrons.com>.
 
-Please have the Thermocycler's serial number available when contacting support. You can find the serial number on the bottom of the module or in the Opentrons App. To find the module serial number in the app, go to the Instruments and Modules section of your robot’s device details page, click the three-dot menu (⋮) and then click **About**.
+Please have the Thermocycler's serial number available when contacting Support. You can find the serial number on the bottom of the module or in the Opentrons App. To find the module serial number in the app, go to the Instruments and Modules section of your robot’s device details page, click the three-dot menu (⋮) and then click **About**.
```
"Opentrons Support" preferred. If we think it's too repetitive, I'd stick with lowercase "support" here.

jwwojak replied:
Using preferred, "Opentrons Support."

ecormany commented on docs/thermocycler-manual/docs/specifications.md:
```diff
@@ -63,48 +64,52 @@ TCG2
-      <td>Solid blue</td>
-      <td>Holding at cool temperature (&lt;23 °C)</td>
+      <td><span class="status-dot-container"><span class="status-dot blue"></span>Blue</span></td>
+      <td>A blue light indicates a cool temperature state: For example:
+        <ul>
+          <li>Solid blue: holding at a cool temperature (&lt;23 °C)</li>
```
Triple colon! I think we can get it down to just one.
```suggestion
      <td>A blue light indicates a cool temperature state.
        <ul>
          <li>Solid blue: holding at a cool temperature (&lt;23 °C)</li>
```
Can also remove "For example:" for red, above.

jwwojak replied:
I can get rid of those multiple : for both heating and cooling.

ecormany reviewed (approved):
Nice batch of final edits. One last polish and this is good to go.

jwwojak committed:
A few final changes

- remove stray punctuation marks.
- add a punctuation mark.
- change the software requirements text back to original, revision made it vague.
- Prose improvements, capitalization
- Remove extra colons from LED table.
